src/main/java/org/opensearch/neuralsearch/processor/chunker/Chunker.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.opensearch.neuralsearch.processor.chunker;
+
+import java.util.Map;
+import java.util.List;
+
+/**
+ * The interface for all chunking algorithms.
+ * All algorithms need to validate parameters and chunk the content.
+ */
+public interface Chunker {
+
+    /**
+     * Validate the parameters for chunking algorithm,
+     * will throw IllegalArgumentException when parameters are invalid
+     *
+     * @param parameters a map containing non-runtime parameters for chunking algorithms
+     */
+    void validateParameters(Map<String, Object> parameters);
+
+    /**
+     * Parse the parameters for chunking algorithm.
+     * The parameters must be validated before parsing.
```

Comment:
Sometimes the parameter has invalid type so that we cannot directly parse these parameters directly. That's why we need the validate parameters before parsing.